manifest-build-tools/application/reprove.py

```diff
@@ -195,6 +214,167 @@ def directory_for_repo(self, repo):
         repo_directory = os.path.join(self._builddir, repo_directory)
         return repo_directory
 
+    def branch_existing_repositories(self):
+        """
+        Issues create branch commands to repos in a provided manifest
+        :return: None
+        """
+        if self._branch_name is None:
+            print "Please provide the new branch name"
+            sys.exit(2)
+
+        repo_list = self._manifest.get_repositories()
+        if repo_list is None:
+            print "No repository list found in manifest file"
+            sys.exit(2)
+        else:
+            # Loop through list of repos and create specified branch on each
+            for repo in repo_list:
+                self.create_repo_branch(repo, self._branch_name)
+
+    def create_repo_branch(self, repo, branch):
+        """
+        create branch  on the repos in the manifest file
+        :param repo: A dictionary
+        :return: None
+        """
+        try:
+            repo_directory = self.directory_for_repo(repo)
```

Comment:
It's "check-out-directory-name" which is a field of repository in manifest.
The field allows a repository to be checked out into a directory which is not the same with the name of repository.